TauAnalysis/MCEmbeddingTools/plugins/DYToElElGenFilter.cc

```diff
@@ -0,0 +1,82 @@
+#include "DataFormats/Candidate/interface/Candidate.h"
+#include "DataFormats/Candidate/interface/LeafCandidate.h"
+#include "DataFormats/HepMCCandidate/interface/GenParticle.h"
+#include "DataFormats/HepMCCandidate/interface/GenParticleFwd.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include "FWCore/Framework/interface/stream/EDFilter.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+class DYToElElGenFilter : public edm::stream::EDFilter<> {
+public:
+  explicit DYToElElGenFilter(const edm::ParameterSet &);
+  ~DYToElElGenFilter() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions &descriptions);
+
+private:
+  void beginStream(edm::StreamID) override;
+  bool filter(edm::Event &, const edm::EventSetup &) override;
+  void endStream() override;
+
+  edm::InputTag inputTag_;
+  edm::EDGetTokenT<reco::GenParticleCollection> genParticleCollection_;
+
+  edm::Handle<reco::GenParticleCollection> gen_handle;
+
+  // virtual void beginRun(edm::Run const&, edm::EventSetup const&) override;
+  // virtual void endRun(edm::Run const&, edm::EventSetup const&) override;
+  // virtual void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+  // virtual void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+
+  // ----------member data ---------------------------
+};
+
+DYToElElGenFilter::DYToElElGenFilter(const edm::ParameterSet &iConfig) {
+  inputTag_ = iConfig.getParameter<edm::InputTag>("inputTag");
+  genParticleCollection_ = consumes<reco::GenParticleCollection>(inputTag_);
+}
+
+DYToElElGenFilter::~DYToElElGenFilter() {}
+
+bool DYToElElGenFilter::filter(edm::Event &iEvent, const edm::EventSetup &iSetup) {
+  iEvent.getByToken(genParticleCollection_, gen_handle);
+
+  for (unsigned int i = 0; i < gen_handle->size(); i++) {
```

Comment:
You can make this loop more efficient by considering the conditions one-by-one and immediately returning false, if they're not met.   Start with ones most likely to be false.  should the cuts be configurable?  For example it's not clear that one wouldn't like to vary the 7 GeV cut.

Reply:
Done in a8eae9f